### PR TITLE
WMTS tile matrices configuration [Server][Feature]

### PR DESCRIPTION
**Pull request by rldhont**
## Description
In QGIS Server 3.4, the WMTS API has been added in which the tile matrices are calculated except fro EPSG:3857 and EPSG:4326.

This commit adds the capability to select the tile matrices to use with the project and to configure the top-left corner and the last level of the tile matrix. In the case of EPSG:3857 and EPSG:4326, the user can only choose the last level of the tile matrices.

Funded by [Ifremer](https://wwz.ifremer.fr/).

## Checklist

- [ ] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and contain sufficient information in the commit message to be documented
- [ ] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [ ] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTING.md#contributing-to-qgis) before each commit

### CODE REVIEW

File: src/app/qgsprojectproperties.cpp [hunk 1]
```diff
@@ -686,6 +687,40 @@ QgsProjectProperties::QgsProjectProperties( QgsMapCanvas *mapCanvas, QWidget *pa
...
+    addWmtsGrid( crsStr );
+  }
+  connect( mWMSList->model(), &QAbstractListModel::rowsInserted, this, &QgsProjectProperties::lwWmsRowsInserted );
+  connect( mWMSList->model(), &QAbstractListModel::rowsRemoved, this, &QgsProjectProperties::lwWmsRowsRemoved );
+  connect( twWmtsGrids, &QTreeWidget::itemDoubleClicked, this, &QgsProjectProperties::twWmtsGridItemDoubleClicked );
+  connect( twWmtsGrids, &QTreeWidget::itemChanged, this, &QgsProjectProperties::twWmtsGridItemChanged );
+  QStringList wmtsGridList = QgsProject::instance()->readListEntry( QStringLiteral( "WMTSGrids" ), QStringLiteral( "CRS" ) );
+  if ( !wmtsGridList.isEmpty() )
+  {
+    QStringList wmtsGridConfigList = QgsProject::instance()->readListEntry( QStringLiteral( "WMTSGrids" ), QStringLiteral( "Config" ) );
```
Comment (nyalldawson) [inline src/app/qgsprojectproperties.cpp]:
```suggestion
    const QStringList wmtsGridConfigList = QgsProject::instance()->readListEntry( QStringLiteral( "WMTSGrids" ), QStringLiteral( "Config" ) );
```

File: src/server/services/wmts/qgswmtsutils.h [hunk 2]
```diff
@@ -41,9 +41,11 @@ namespace QgsWmts
 
     QgsRectangle extent;
 
+    QgsUnitTypes::DistanceUnit unit;
```
Comment (nyalldawson) [inline src/server/services/wmts/qgswmtsutils.h]: initialise to some default value

**Review by elpaso — APPROVED** [review]
Cool stuff!

**Comment by Abelon38** [comment]
Hi gays!

I hope you are doing well! I just installed the new QGIS server 3.10.2 with PROJ 6.3.0 on Windows and everything runs well, but not the WMTS. I can only reproduce WMTS in SRS 3857. I know  that it has been discussions related an issue with PROJ 6. 
I wonder if I should try with PROJ 5 in order to get my WMTS to work in other SRS than 3857?

Thanks for your time!

BR/Abel

**Pull request merged**